app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
+  end
+
+  def loa_no_js_fallback_link
+    if loa1_signup_context?
+      destroy_user_path
+    elsif loa3_context?
+      verify_session_path
+    end
```

Comment:
This logic does not match the logic when JS is enabled. With this logic, a user that has JS disabled and creates an account via an LOA3 request will not be able to cancel their account on the OTP verification page because `loa_no_js_fallback_link` will be `nil`, and the `shared/cancel` partial will fallback to `link`, which on the OTP verification page is `/api/saml/logout`, which signs the user out.